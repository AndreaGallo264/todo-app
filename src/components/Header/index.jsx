import React from "react";
import { useDispatch } from "react-redux";
import { AppBar, Typography, Box, Button } from "@material-ui/core";
import { useLocation, Link } from "react-router-dom";
import { removeCompletedTasks } from "../../store/actions";

const Header = () => {
  let location = useLocation();
  const dispatch = useDispatch();
  return (
    <AppBar position="fixed">
      <Box
        display="flex"
        justifyContent="space-between"
        flexWrap="wrap"
        padding={2}
      >
        <Link to="/">
          <Typography variant="h3" component="h2" color="secondary">
            To do App
          </Typography>
        </Link>
        {location.pathname === "/" && (
          <Button
            onClick={() => {
              dispatch(removeCompletedTasks());
            }}
            variant="contained"
            color="secondary"
          >
            Filter completed tasks
          </Button>
        )}
      </Box>
    </AppBar>
  );
};

export default Header;
