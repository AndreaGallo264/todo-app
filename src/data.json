[{
  "id": 1,
  "task": "Give dog a bath",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 2,
  "task": "Do laundry",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 3,
  "task": "Vacuum floor",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 4,
  "task": "Feed cat",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 5,
  "task": "Change light bulbs",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 6,
  "task": "Go to Store",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 7,
  "task": "Fill gas tank",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 8,
  "task": "Change linens",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 9,
  "task": "Rake leaves",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 10,
  "task": "Bake Cookies",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 11,
  "task": "Take nap",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 12,
  "task": "Read book",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 13,
  "task": "Exercise",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 14,
  "task": "Give dog a bath",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 15,
  "task": "Do laundry",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 16,
  "task": "Vacuum floor",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 17,
  "task": "Feed cat",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 18,
  "task": "Change light bulbs",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 19,
  "task": "Go to Store",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 20,
  "task": "Fill gas tank",
  "complete": false,
  "dueDate": "2021-05-08"
}]