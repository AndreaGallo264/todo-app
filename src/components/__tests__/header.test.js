import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import data from "../../data.json";
import Header from "../Header";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/",
  }),
}));

describe("Header component", () => {
  test("Matches snapshot", () => {
    const mockStore = configureMockStore();
    const store = mockStore({ list: data, lastId: data.length });
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
