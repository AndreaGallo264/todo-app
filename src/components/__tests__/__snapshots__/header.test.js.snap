// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h2
        className="MuiTypography-root MuiTypography-h3 MuiTypography-colorSecondary"
      >
        To do App
      </h2>
    </a>
  </div>
</header>
`;
